src/screens/DonateScreen/NumberTextField.js

```diff
@@ -35,25 +72,52 @@ class DonateScreen extends React.PureComponent<Props, State> {
   };
 
   render() {
-    const { placeholder, onFocus, ...otherProps } = this.props;
+    const { label, placeholder, onFocus, style, ...otherProps } = this.props;
     const { focused } = this.state;
+
+    // Note on accessibility:
+    // React Native does not support to mark up labels for input fields
+    // properly. See for example: https://github.com/facebook/react-native/issues/14989
+
     return (
-      <TextInput
-        keyboardType="numeric"
-        onBlur={this.onBlur}
-        onFocus={this.onFocus}
-        placeholder={focused ? placeholder : null}
-        placeholderTextColor={mediumGreyColor}
-        style={[styles.base, focused && styles.focused]}
-        underlineColorAndroid={transparent}
-        {...otherProps}
-      />
+      <View
+        accessible
+        accessibilityLabel={getAccessibiltyLabel(
+          label,
+          otherProps.value,
+          focused
+        )}
+        style={style}
+      >
+        <Text
+          type="h4"
+          color="lightNavyBlueColor"
+          importantForAccessibility="no"
+          style={styles.label}
+        >
+          {label}
+        </Text>
+        <TextInput
+          keyboardType="numeric"
+          onBlur={this.onBlur}
+          onFocus={this.onFocus}
+          placeholder={focused ? placeholder : null}
+          placeholderTextColor={mediumGreyColor}
+          style={[styles.input, focused && styles.inputFocused]}
+          underlineColorAndroid={transparent}
+          importantForAccessibility="no"
```

Comment:
For some reason the TextInput works much better on TalkBack, when you hide it from screen readers completely and put in inside a view, which is accessible.